tests/test_posixtestsuite.py

```diff
@@ -0,0 +1,3466 @@
+               expected = '0',
+               args = posix_tests_common_args + [
+                   '-s', 'USE_PTHREADS=1',
+                   '-s', 'PTHREAD_POOL_SIZE=8',
+                   '-I', posix_test_dir('pthread_attr_getdetachstate')])
+
+
+  @requires_threads
+  def test_pthread_attr_getinheritsched_1_1(self):
+    self.btest(posix_test_file('pthread_attr_getinheritsched', '1-1.c'),
+               expected = '0',
+               args = posix_tests_common_args + [
+                   '-s', 'USE_PTHREADS=1',
+                   '-s', 'PTHREAD_POOL_SIZE=8',
+                   '-I', posix_test_dir('pthread_attr_getinheritsched')])
+
+
+  @requires_threads
+  def test_pthread_attr_getschedparam_1_1(self):
+    self.btest(posix_test_file('pthread_attr_getschedparam', '1-1.c'),
+               expected = '0',
+               args = posix_tests_common_args + [
+                   '-s', 'USE_PTHREADS=1',
+                   '-s', 'PTHREAD_POOL_SIZE=8',
+                   '-I', posix_test_dir('pthread_attr_getschedparam')])
+
+
+  @requires_threads
+  def test_pthread_attr_getschedpolicy_2_1(self):
+    self.btest(posix_test_file('pthread_attr_getschedpolicy', '2-1.c'),
+               expected = '0',
+               args = posix_tests_common_args + [
+                   '-s', 'USE_PTHREADS=1',
+                   '-s', 'PTHREAD_POOL_SIZE=8',
+                   '-I', posix_test_dir('pthread_attr_getschedpolicy')])
+
+
+  @requires_threads
+  def test_pthread_attr_getscope_1_1(self):
+    self.btest(posix_test_file('pthread_attr_getscope', '1-1.c'),
+               expected = '0',
+               args = posix_tests_common_args + [
+                   '-s', 'USE_PTHREADS=1',
+                   '-s', 'PTHREAD_POOL_SIZE=8',
+                   '-I', posix_test_dir('pthread_attr_getscope')])
+
+
+  @requires_threads
+  def test_pthread_attr_getstack_1_1(self):
+    self.btest(posix_test_file('pthread_attr_getstack', '1-1.c'),
+               expected = '0',
+               args = posix_tests_common_args + [
+                   '-s', 'USE_PTHREADS=1',
+                   '-s', 'PTHREAD_POOL_SIZE=8',
+                   '-I', posix_test_dir('pthread_attr_getstack')])
+
+
+  @requires_threads
+  def test_pthread_attr_getstacksize_1_1(self):
+    self.btest(posix_test_file('pthread_attr_getstacksize', '1-1.c'),
+               expected = '0',
+               args = posix_tests_common_args + [
+                   '-s', 'USE_PTHREADS=1',
+                   '-s', 'PTHREAD_POOL_SIZE=8',
+                   '-I', posix_test_dir('pthread_attr_getstacksize')])
+
+
+  @requires_threads
+  def test_pthread_attr_init_1_1(self):
+    self.btest(posix_test_file('pthread_attr_init', '1-1.c'),
+               expected = '0',
+               args = posix_tests_common_args + [
+                   '-s', 'USE_PTHREADS=1',
+                   '-s', 'PTHREAD_POOL_SIZE=8',
+                   '-I', posix_test_dir('pthread_attr_init')])
```

Comment:
There are so many of these I'm almost temped to make these data driven. 

```
for t in test_array:
   add_test_method(t.....)
```

But thats even more obscure so maybe its not worth it.

Reply:
Done in 4c4fd41